runtime/program_params_validation_test.go

```diff
@@ -1391,4 +1391,86 @@ func TestRuntimeTransactionParameterTypeValidation(t *testing.T) {
 		expectRuntimeError(t, err, &ArgumentNotImportableError{})
 	})
 
+	newEnumType := func() cadence.Enum {
+		return cadence.NewEnum([]cadence.Value{
+			cadence.NewInt(0),
+		}).WithType(cadence.NewEnumType(
+			common.AddressLocation{
+				Address: common.MustBytesToAddress([]byte{0x1}),
+				Name:    "C",
+			},
+			"C.Alpha",
+			cadence.IntType,
+			[]cadence.Field{
+				{
+					Identifier: sema.EnumRawValueFieldName,
+					Type:       cadence.IntType,
+				},
+			},
+			nil,
+		))
+	}
+
+	t.Run("Enum Optional Type", func(t *testing.T) {
+		t.Parallel()
+
+		contracts := map[common.AddressLocation][]byte{
+			{
+				Address: common.MustBytesToAddress([]byte{0x1}),
+				Name:    "C",
+			}: []byte(`
+		   access(all) contract C {
+			   access(all) 
+			   enum Alpha: Int {
+				  access(all)
+				  case A
+
+				  access(all)
+				  case B
+			  }
+		   }
+		`),
+		}
+
+		script := `
+	  import C from 0x1
+
+	  transaction(arg: C.Alpha?) {}
+	`
+
+		err := executeTransaction(t, script, contracts, cadence.NewOptional(nil))
+		assert.NoError(t, err)
+	})
+
+	t.Run("Enum Type", func(t *testing.T) {
+		t.Parallel()
+
+		contracts := map[common.AddressLocation][]byte{
+			{
+				Address: common.MustBytesToAddress([]byte{0x1}),
+				Name:    "C",
+			}: []byte(`
+		   access(all) contract C {
+			   access(all) 
+			   enum Alpha: Int {
+				  access(all)
+				  case A
+
+				  access(all)
+				  case B
+			  }
+		   }
+		`),
+		}
+
+		script := `
+	  import C from 0x1
+
+	  transaction(arg: C.Alpha) {}
```

Comment:
Maybe extend the body here a bit and use `arg`, e.g. by comparing it, logging it, etc.

Reply:
Done.